### PR TITLE
[layout] adapt javadoc to API changes

### PR DESCRIPTION
**Pull request by JanKoehnlein**

### CODE REVIEW

File: org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java [hunk 1]
```diff
@@ -237,7 +237,7 @@ public void setNeedsClientLayout(boolean value) {
 	 * with the layout engine configured with {@link #setLayoutEngine(ILayoutEngine)}, so returning {@code true}
 	 * here makes sense only if such an engine is available.
 	 * 
-	 * <p>The default implementation returns the value configured with {@link #setNeedsServerLayout(boolean)},
+	 * <p>The default implementation returns the value configured with {@link #getServerLayoutKind(ServerLayoutKind)},
```
Comment (spoenemann) [inline org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java]: This should point to `setServerLayoutKind`.

File: org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/ILayoutEngine.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 /**
  * A layout engine is able to compute layout information for a model. Invoked by {@link DefaultDiagramServer}
- * when {@link DefaultDiagramServer#needsServerLayout(SModelRoot)} returns {@code true}.
+ * when {@link DefaultDiagramServer#getServerLayoutKind(SModelRoot)} returns {@code true}.
```
Comment (spoenemann) [inline org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/ILayoutEngine.java]: The method no longer returns `boolean`, but an enum.

File: org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java (hunk 1 above)
Comment (JanKoehnlein) [reply org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java]: Fixed

File: org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/ILayoutEngine.java (hunk 2 above)
Comment (JanKoehnlein) [reply org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/ILayoutEngine.java]: Fixed

**Pull request merged**